src/idx_html.pre.js

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2019 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+/* eslint-disable import/no-unresolved */
+const htmlPre = require('./html.pre');
+
+/**
+ * The 'pre' function that is executed before the HTML is rendered
+ * @param context The current context of processing pipeline
+ * @param context.content The content
+ */
+function pre(context) {
+  const { document } = context.content;
+  htmlPre.pre(context);
+
+  // construct the tables
+  const tables = [];
+  const basic = { name: 'basic', entries: {} };
+  const images = { name: 'images', entries: {} };
+
+  const titleEl = document.querySelector('h1');
+  if (titleEl) {
+    basic.entries.title = titleEl.textContent;
+  }
+
+  const descEl = document.querySelector('.title .header p');
+  if (descEl) {
+    basic.entries.description = descEl.textContent;
+  }
+
+  const imgs = [];
+  document.querySelectorAll('img').forEach((img) => {
+    imgs.push(img.src);
+  });
+  images.entries = { images: imgs };
+  tables.push(basic);
+  tables.push(images);
+  context.content.json = { tables };
+
+  context.content.json.string = JSON.stringify(context.content.json);
```

Comment:
Minor details, but... I think adding a `string` property to an object to store its stringified version is a bit... strange ;) Also, you do not seem to need the json object.
why not just:

```
context.content.tables = JSON.stringify(tables);
```

or at least:

```
context.content.tables = { 
  json: tables,
  str: JSON.stringify(tables); 
};
```

Reply:
i see.. i am currently using json obj as part of my request in my node app, if we use this then the header would be `text/html` and i would need to do `JSON.parse` but let me look into it and get back to you.